### PR TITLE
Expose more syncing types to enable custom syncing strategy

### PR DESCRIPTION
**Pull request by liuchengxu**
This PR exposes additional syncing types to facilitate the development of a custom syncing strategy based on the existing Polkadot syncing strategy. Specifically, my goal is to isolate the state sync and chain sync components, allowing the state to be downloaded from the P2P network without running a full regular Substrate node. I also need to intercept the state responses during the state sync process.

The newly exposed types are necessary to implement this custom syncing strategy.

### CODE REVIEW

**Comment by liuchengxu** [comment]
@bkchr @dmitry-markin Could you please take a look at this tiny PR? It should be real quick. All types exposed here are needed in my custom syncing strategy.

File: substrate/client/network/sync/src/strategy/state.rs [hunk 1]
```diff
@@ -124,9 +124,7 @@ impl<B: BlockT> StateStrategy<B> {
 	}
 
 	// Create a new instance with a custom state sync provider.
-	// Used in tests.
```
Comment (lexnv) [inline substrate/client/network/sync/src/strategy/state.rs:127]: nit: Could we add a comment to mention that most of the times users are looking for `StateStrategy::new` and this should be used by custom strategies and advanced developers?

**Review by lexnv — APPROVED** [review]
Thanks for contributing!

**Review by dmitry-markin — APPROVED** [review]

**Comment by liuchengxu** [comment]
@lexnv @dmitry-markin Anything missing to merge this?

**Pull request merged**